File: golang/smily.go
package main

func CountSmilyFace(smileys []string) int {
	typeSmileys := map[string]bool{
		":~)": true,
		":-)": true,
		":~D": true,
		":-D": true,
		":)":  true,
		":D":  true,
		";-)": true,
		";~)": true,
		";~D": true,
		";-D": true,
		";)":  true,
		";D":  true,
	}
	total := 0

	for _, smile := range smileys {
		if _, ok := typeSmileys[smile]; ok {
			total++
		}
	}

	return total
}

File: javascript/smily.js
/**
 * 
 * @param {string[]} texts list of string [":)", ":(", ":>"]
 * @returns {number} amount of smily face detected
 */
const countSmileyFace = (arrSmileys) => {
    if (arrSmileys.length <= 0) return 0

    const typeSmileys = [':~)', ':-)', ':~D', ':-D', ':)', ':D', ';-)', ';~)', ';~D', ';-D', ';)', ';D']

    let total = 0
    for (let i in arrSmileys) {
        const smile = arrSmileys[i]
        if (typeSmileys.some((f) => f === smile)) {
            total++
        }
    }
    return total
}

module.exports = {
    countSmileyFace
}